allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -110,10 +110,11 @@ class DatasetReader(Registrable):
         :func:`_instances_from_cache_file`).  If the cache file does _not_ exist, we will _create_
         it on our first pass through the data (using :func:`_instances_to_cache_file`).
 
-        IMPORTANT CAVEAT: It is the _caller's_ responsibility to make sure that this directory is
-        unique for any combination of code and parameters that you use.  That is, if you pass a
-        directory here, we will use any existing cache files in that directory _regardless of the
-        parameters you set for this DatasetReader!_
+        !!! NOTE
+            It is the _caller's_ responsibility to make sure that this directory is
+            unique for any combination of code and parameters that you use.  That is, if you pass a
+            directory here, we will use any existing cache files in that directory _regardless of the
+            parameters you set for this DatasetReader!_
```

Comment:
We could hash the parameters to the dataset reader and make the hash part of the filename. But that's for another PR.

Reply:
That's a good thought.